src/main/java/org/jenkinsci/test/acceptance/junit/WithCredentials.java

```diff
@@ -43,6 +44,7 @@
 @Target({METHOD, TYPE})
 @Inherited
 @Documented
+@WithPlugins("credentials@2.0.7")
```

Comment:
Your motivation makes sense, though the reason why I and Jesse are surprised is that noone _implemented_ that to work. Unfortunately, what you are observing is an accidental behavior - ATH just installs latest version regardless of your annotation here. If you change the annotation to `@WithPlugins("credentials@3.0.7")` (version that does not exist), it installs 2.0.7 (latest) instead of failing the test.

Throwing `AssumptionViolatedException` from Statement#evaluate is the way to go.